sim/battle.ts

```diff
@@ -1020,12 +1032,34 @@ export class Battle extends Dex.ModdedDex {
 			type = this.requestState;
 		}
 
-		// default to no request
-		const requests: any[] = Array(this.sides.length).fill(null);
 		for (const side of this.sides) {
 			side.activeRequest = null;
 		}
 
+		const maxTeamSize = this.getMaxTeamSize();
+		if (type === 'teampreview') {
+			this.add('teampreview' + (maxTeamSize !== 6 ? '|' + maxTeamSize : ''));
```

Comment:
... I had specifically changed this in PR #4281, and you went and changed it back again...

Reply:
Probably you should just change it again, but with a comment this time. :p